WordPress/gutenberg
# Components: Consolidate and add docs to Storybook

## What?
Consolidate multiple instances of key docs (`README.md` and `CONTRIBUTING.md`) and configure Storybook to display them.

## Why?
Up until now we've had two copies of both `README` and `CONTRIBUTING`: one in the package root, and another in the `src` directory. Consolidating these makes things clearer for contributors and surfacing them in Storybook makes them more visible and easier to access. Also because it was fun.

## How?
The two copies of `CONTRIBUTING` were identical, so the duplicate was just deleted.

The contents of the two `README` files were different, with the root copy focusing on detailed instructions for different aspects of a contribution, while the `src` copy focused more on the workflow of introducing a new component. I've combined all of this into a single file with one small change. The `src` copy's workflow for new components included a step to create a [Figma component](https://www.figma.com/file/ZtN5xslEVYgzU7Dd5CxgGZwq/WordPress-Components), which I don't believe we're still doing, so I took that step out. It's easy enough to add back in if it's still relevan.

Finally to render both of these docs in Storybook, I enabled the `transcludeMarkdown` config setting, which allows mdx files to import `.md` files and render their contents as a component. I then added two new `.mdx` files, one to import each of the desired docs.

## Testing Instructions
Look over both affected docs to make sure the relevant info has been carried over. Check that the both render nicely in Storybook.
